### PR TITLE
Various improvements to flockeropenssl scripts

### PR DESCRIPTION
**Pull request by sgnn7**
https://phabricator.endlessm.com/T11072

### CODE REVIEW

**Comment by sgnn7** [comment]
@wallnerryan Tested on an internal cluster with all the changes and things seem to be running smoothly.

File: README.md [hunk 1]
```diff
@@ -2,59 +2,34 @@
...
-```
-
-> Note:  You can edit `dir = ` under `[ CA_default ]` to either `$HOME/ssl` from the above example or a custom location you are using for your root ssl directory.
-
 #### Generate Flocker Certificates
 
 You can view help message by
 ```
 $ ./flockeropenssl/generate_flocker_certs.sh -h
+Usage: $0 (-i=<control_ip> | -d=<control_fqdn>) [-f=openssl_conf] -c=<cluster_name> -n=<node>[,<node> ... ]
```
Comment (wallnerryan) [inline README.md:21]: thanks this is cleaner

File: generate_flocker_certs.sh [hunk 2]
```diff
@@ -39,133 +39,256 @@ HELP_MSG="""
...
+  echo "ERROR! No control service FQDN or IP provided! Exiting!"
+  echo "${HELP_MSG}"
+  exit 1
 fi
 
-# ----------------------------------------
-# generate the CA:
-# ----------------------------------------
+echo "Cleaning up old CA dirs"
+rm -rf ssl
```
Comment (wallnerryan) [inline generate_flocker_certs.sh]: removing things makes me nervous, especially if someone uses "ssl" outside of this script. Which could be the case if its not `./ssl` Maybe rename this to `./flockerssl` so its obvious.  Same in the openssl conf file.

File: README.md [hunk 3]
```diff
@@ -63,51 +38,34 @@ $ ./flockeropenssl/generate_flocker_certs.sh -h
 -h | --help (This help message)
 ```
 
-Example
+Examples:
 
 ```
-./flockeropenssl/generate_flocker_certs.sh -d="ec2-52-91-11-106.compute-1.amazonaws.com" -n="ec2-52-91-11-106.compute-1.amazonaws.com,node2,node3" -f=/etc/flocker/ssl/flockeropenssl/openssl.cnf
+./flockeropenssl/generate_flocker_certs.sh -d=www.foobar.com -k=1024 -c=staging-us-west-2 -n=one,two
```
Comment (wallnerryan) [inline README.md]: make this more generic, instead of amazon-like azs. "staging-1" for instance.

File: generate_flocker_certs.sh [hunk 4]
```diff
@@ -39,133 +39,256 @@ HELP_MSG="""
...
+
+output_dir=${OUTPUT_DIR:="$CURRENT_DIR/clusters/$cluster_name"}
+echo "Output directory is $output_dir"
+
+if [ -d $output_dir ]; then
+  if [ "$FORCE_OVERWRITE" == "true" ]; then
+    echo "Forcefuly removing old data from old cluster"
+    rm -rf "$output_dir"
+  else
+    echo "ERROR! Cluster already created and '--force' not specified! Exiting!"
```
Comment (wallnerryan) [inline generate_flocker_certs.sh:238]: thanks, this is nice.

**Comment by wallnerryan** [comment]
@sgnn7  Looks great, thanks, few comments. Also we're moving this to https://github.com/ClusterHQ/flocker-openssl.  can you make the above changes and re-submit to the new locations, then close this PR.

File: generate_flocker_certs.sh [hunk 5]
```diff
@@ -1,36 +1,36 @@
...
-# to point at your ssl directory.
-#
-# most of this was taken from this tutorial:
+################################################################################
+# Most of this was taken from this tutorial:
 # https://jamielinux.com/docs/openssl-certificate-authority/index.html
 # and combined with the implementation in flocker/flocker/ca/_ca.py
 #
-# authors: Brendan Cox "justnoise" , and Ryan Wallner "wallnerryan"
+# Authors:
```
Comment (wallnerryan) [inline generate_flocker_certs.sh]: mind splitting this out into CONTRIBUTORS.md?

**Comment by sgnn7** [comment]
@wallnerryan Closing PR to address changes and to reopen it for new repo (https://github.com/ClusterHQ/flocker-openssl)

**Pull request closed without merging**